src/h5web/nexus-visualizer/NexusVisualizer.tsx

```diff
@@ -1,17 +1,73 @@
 import React, { ReactElement } from 'react';
 import type { HDF5Group } from '../providers/models';
+import { getAttributeValue, getInterpretationVis } from './utils';
 import styles from '../Visualizer.module.css';
+import { useEntity } from '../providers/hooks';
+import { isDataset } from '../providers/utils';
+import VisSelector from '../dataset-visualizer/VisSelector';
+import VisDisplay from '../dataset-visualizer/VisDisplay';
+import { VIS_DEFS } from '../visualizations';
+import { getSupportedVis } from '../dataset-visualizer/utils';
+import { NXDataAttribute } from './models';
 
 interface Props {
   group: HDF5Group;
 }
 
 function NexusVisualizer(props: Props): ReactElement {
   const { group } = props;
+
+  const signalDatasetTitle = getAttributeValue(group, NXDataAttribute.Signal);
+  const signalLink =
+    signalDatasetTitle &&
+    group.links?.find((l) => l.title === signalDatasetTitle);
+  const signalDataset = useEntity(signalLink);
+
+  if (signalDatasetTitle === undefined) {
```

Comment:
If `signalDatasetTitle` is `""`, the error is still relevant, isn't it? So `!signalDatasetTitle ` might do.

Reply:
No as in this case, `signalDatasetTitle` exists ! So it should be caught by the second guard clause.